src/bsk_rl/gym.py

```diff
@@ -507,17 +507,27 @@ def reset(
     ) -> tuple[MultiSatObs, dict[str, Any]]:
         """Reset the environment and return PettingZoo Parallel API format."""
         self.newly_dead = []
+        self._agents_last_compute_time = None
         return super().reset(seed, options)
 
     @property
     def agents(self) -> list[AgentID]:
         """Agents currently in the environment."""
-        truncated = super()._get_truncated()
-        return [
-            satellite.name
-            for satellite in self.satellites
-            if (satellite.is_alive() and not truncated)
-        ]
+        if (
+            self._agents_last_compute_time is None
+            or self._agents_last_compute_time != self.simulator.sim_time
+        ):
+            truncated = super()._get_truncated()
+            agents = [
+                satellite.name
+                for satellite in self.satellites
+                if (satellite.is_alive() and not truncated)
```

Comment:
Truncated is just one boolean for the episode and not for each satellite, right? I guess you could just check if it is truncated before iterating over the satellites to make it faster? Still, I am not sure if this has any practical advantages.

Reply:
I think this was copying some example from petting zoo. I don’t see that being a huge performance difference.